### PR TITLE
Marked classes and interfaces as API part 2

### PR DESCRIPTION
**Pull request by ProkopovVitaliy**
### Fixed Issues (if relevant)

1. Fixes magento/magento2#32056

### Manual testing scenarios (*)

1. ...
2. ...

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by ProkopovVitaliy** [comment]
@magento run all tests

**Comment by ProkopovVitaliy** [comment]
@magento run test Static Tests, Functional Tests CE

**Comment by ProkopovVitaliy** [comment]
@magento run tests Static Tests, Functional Tests CE

**Comment by ProkopovVitaliy** [comment]
@magento run Static Tests, Functional Tests CE

**Review by eduard13 — CHANGES_REQUESTED** [review]
@ProkopovVitaliy please mark the non-public types as `api` as well, which should fix the Static Tests.
Thank you.

**Comment by eduard13** [comment]
magento run Database Compare

**Comment by eduard13** [comment]
@magento run Database Compare

**Comment by eduard13** [comment]
Hi @ProkopovVitaliy will you be able to update this PR, in order to fix the Static Tests?

**Comment by ProkopovVitaliy** [comment]
@magento run Static Tests

**Comment by ProkopovVitaliy** [comment]
@magento run Static Tests

**Comment by ProkopovVitaliy** [comment]
@magento run all tests

File: app/code/Magento/Customer/Block/Adminhtml/Edit/GenericButton.php [hunk 1]
```diff
@@ -10,6 +10,7 @@
 /**
  * Class GenericButton
  * @package Magento\Customer\Block\Adminhtml\Edit
```
Comment (eduard13) [inline app/code/Magento/Customer/Block/Adminhtml/Edit/GenericButton.php]: Let's remove the `@package` annotation.
```suggestion
```

File: lib/internal/Magento/Framework/Console/Cli.php [hunk 2]
```diff
@@ -30,6 +30,7 @@
  *
  * @inheritdoc
  * @SuppressWarnings(PHPMD.CouplingBetweenObjects)
+ * @api
```
Comment (eduard13) [inline lib/internal/Magento/Framework/Console/Cli.php:31]: Usually, the `@api` annotation should be followed by other annotations, like `@SuppressWarnings`. Lets remove the `@inheritdoc` and move `@SuppressWarnings` at the end, separated by a new line.

File: app/code/Magento/Catalog/Controller/Adminhtml/Product.php [hunk 3]
```diff
@@ -10,6 +10,7 @@
 /**
  * Catalog product controller
  * @SuppressWarnings(PHPMD.NumberOfChildren)
+ * @api
```
Comment (eduard13) [inline app/code/Magento/Catalog/Controller/Adminhtml/Product.php:15]: Lets move the `@SuppressWarnings` at the end, and `@api` should follow class description.

**Review by eduard13 — CHANGES_REQUESTED** [review]
Thank you @ProkopovVitaliy for fixing the Static Tests, could you please also fix these minor things I've mentioned below?

**Comment by eduard13** [comment]
@magento run Functional Tests B2B, Database Compare

**Comment by ProkopovVitaliy** [comment]
@magento run all tests

**Comment by eduard13** [comment]
@magento run Database Compare

**Comment by eduard13** [comment]
@magento run Database Compare, Functional Tests B2B

**Comment by eduard13** [comment]
@magento run Database Compare, Functional Tests B2B

**Comment by eduard13** [comment]
@magento run Database Compare, Functional Tests B2B

**Comment by ProkopovVitaliy** [comment]
@magento run all tests

**Comment by ProkopovVitaliy** [comment]
@magento run WebAPI Tests

**Review by eduard13 — APPROVED** [review]
Good job 👍

**Comment by magento-engcom-team** [comment]
Hi @eduard13, thank you for the review.
[ENGCOM-8930](https://jira.corp.magento.com/browse/ENGCOM-8930) has been created to process this Pull Request

**Comment by gabrieldagama** [comment]
Hi @ProkopovVitaliy can you recreate this PR to the api-changes-develop branch instead? Just make sure to create a new branch from this one and cherry pick your commit.

Thanks!

**Comment by gabrieldagama** [comment]
Closed in favor of https://github.com/magento/magento2/pull/33222

**Pull request closed without merging**